### PR TITLE
bug fix in os wrapper

### PR DESCRIPTION
**Pull request by Maoni0**
@janvorli PTAL. needed to add the casting to size_t part otherwise it's treated as 32-bit int.

### CODE REVIEW

**Comment by janvorli** [comment]
The Unix builds are failing due to the removal of the #ifndef FEATURE_CORECLR. The affinity setting is not available on Unix.

**Comment by Maoni0** [comment]
@dotnet-bot test this please

File: src/vm/gcenv.os.cpp [hunk 1]
```diff
@@ -636,24 +636,22 @@ bool GCToOSInterface::CreateThread(GCThreadFunction function, void* param, GCThr
...
-        ga.Mask = 1 << affinity->Processor;
+        ga.Mask = (size_t)1 << affinity->Processor;
 
         CPUGroupInfo::SetThreadGroupAffinity(gc_thread, &ga, NULL);
     }
-    else if (affinity->Processor != -1)
+    else if (affinity->Processor != GCThreadAffinity::None)
     {
-        SetThreadAffinityMask(gc_thread, 1 << affinity->Processor);
+        SetThreadAffinityMask(gc_thread, (size_t)1 << affinity->Processor);
```
Comment (mikedn) [inline src/vm/gcenv.os.cpp]: `DWORD_PTR` would be better as that's what `SetThreadAffinityMask` uses.

Comment (Maoni0) [inline src/vm/gcenv.os.cpp]: good point...I'll do that. thanks!

**Comment by Maoni0** [comment]
@dotnet-bot test this please

**Comment by Maoni0** [comment]
@dotnet-bot test Windows_NT x86 Release Legacy Backend Build and Test

**Comment by janvorli** [comment]
@dotnet-bot test FreeBSD x64 Checked Build please

**Comment by janvorli** [comment]
@dotnet-bot test OSX x64 Checked Build and Test please

**Comment by Maoni0** [comment]
@dotnet-bot test OSX x64 Checked Build and Test

**Comment by Maoni0** [comment]
@dotnet-bot test Ubuntu arm Cross Release Build

**Comment by Maoni0** [comment]
spoke to @ramarag and @mmitche , I can ignore the arm build and merge.

**Pull request merged**